runner_benchmark/utils.py

```diff
@@ -18,4 +18,10 @@ def parse_params_from_location(url, route):
             if field_pos == value_pos:
                 params[field_name] = value
 
+    if urlparse(route).query:
+        query_result = urlparse(route).query.partition("=")[2]
+        url_query = urlparse(url).query.partition("=")[2]
+
+        params[query_result.strip("{}")] = url_query
+
```

Comment:
- `urlparse` should be called once for each url
- this only works for one query parameter, we should look at using the `parse_qs` method from `urlparse` and make it work for any number of query parameters.

Reply:
I've refactored this to use the parse_qs method now but think it still might be able to be shortened as wasn't sure what to change for your first point?

Reply:
- `urlparse` is already called on lines 5 and 6. Those lines should be changed to store the parsed url, something like:

    ```python
    parsed_route = urlparse(route)
    parsed_url = urlparse(url)
    ```

    Then the path can be accessed with `parsed_route.path` and the query with `parsed_url.query`.

- When looping over the query params, can this be simplified to something like:

    ```python
    params.update((key, url_result[key][0]) for key in query_result)
    ```

- I think we might be able to use `parse_qsl` rather than `parse_qs` and remove the `[0]` access.